src/schema/users.ts

```diff
@@ -1256,6 +1271,17 @@ export const resolvers: IResolvers<any, Context> = {
         );
       return getCurrentUser(ctx, info);
     },
+    addUserAcquisitionChannel: async (
+      _,
+      { source }: { source: string },
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      await ctx.con
+        .getRepository(User)
```

Comment:
I would validate that `source` here is one of keys of `AcquisitionChannel`. We don't want any garbage there right?

Reply:
Also maybe name the argument `acquisitionChannel` because `source` means something else (eg. our entity) in different mutations.

Reply:
Ah, source is validate through graphql schema, so just renaming potentially.

Reply:
Yes. I just used `source` as it was defined in the RFC GraphQL schema as that.

Reply:
This actually makes me think whether we can just utilize the updateUserProfile, but I went and did what RFC have written.

Reply:
Yeah I think it can still be changed, just so it does not look as if it should be our source (as our entity) as this is something else.

Reply:
We could use the `updateUserProfile`, but IMO this value has nothing to do with user profile, hence why I suggested a separate mutation.